Adds a section on input validation (#2265)
buildkite/docs: pages/pipelines/input_step.md

            value: "stable"
```
{: codeblock-file="pipeline.yml"}

## Input validation

> To prevent the users from entering invalid values when using an input step with text inputs (for example, to gather some deployment information), you can use input validation.

Reviewer: I think this could be a little tighter, but happy for you to leave it as-is if you think it's clearer:

```suggestion
To prevent users from entering invalid text values in input steps  (for example, to gather some deployment information), you can use input validation.
```

> If you associate a RegEx to a field, the field outline will turn red when an invalid value is entered.

Reviewer: Let's not abbreviate this:

```suggestion
If you associate a regular expression to a field, the field outline will turn red when an invalid value is entered.
```

To do it, use the following sample syntax:

```yml
steps:
  - input: "Click me!"
    fields:
      - text: Must be hexadecimal
        key: hex
        format: "[0-9a-f]+"
```

> The format must be a regular expression implicitly anchored to the beginning and end of the input and is functionally equivalent to the [HTML5 pattern attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern).

Reviewer: Use code formatting when referring to the field:

```suggestion
The `format` must be a regular expression implicitly anchored to the beginning and end of the input and is functionally equivalent to the [HTML5 pattern attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern).
```
